ui/components/src/ThresholdsEditor/ThresholdsEditor.tsx

```diff
@@ -0,0 +1,236 @@
+  const {
+    thresholds: { defaultColor, palette },
+  } = useChartsTheme();
+  const defaultThresholdColor = thresholds?.default_color ?? defaultColor;
+
+  const [steps, setSteps] = useState(thresholds?.steps);
+  useEffect(() => {
+    setSteps(thresholds?.steps);
+  }, [thresholds?.steps]);
+
+  // every time a new threshold is added, we want to focus the recently added input
+  const recentlyAddedInputRef = useRef<HTMLInputElement | null>(null);
+  const focusRef = useRef(false);
+  useEffect(() => {
+    if (!recentlyAddedInputRef.current || !focusRef.current) return;
+    recentlyAddedInputRef.current?.focus();
+    focusRef.current = false;
+  }, [steps?.length]);
+
+  const handleThresholdChange = (e: React.ChangeEvent<HTMLInputElement>, i: number) => {
+    setSteps(
+      produce(steps, (draft) => {
+        const step = draft?.[i];
+        if (step) {
+          step.value = Number(e.target.value);
+        }
+      })
+    );
+  };
+
+  const handleThresholdColorChange = (color: string, i: number) => {
+    if (thresholds !== undefined) {
+      onChange(
+        produce(thresholds, (draft) => {
+          if (draft.steps !== undefined) {
+            const step = draft.steps[i];
+            if (step) {
+              step.color = color;
+            }
+          }
+        })
+      );
+    }
+  };
+
+  const handleDefaultColorChange = (color: string) => {
+    if (thresholds !== undefined) {
+      onChange(
+        produce(thresholds, (draft) => {
+          draft.default_color = color;
+        })
+      );
+    }
+  };
+
+  // sort thresholds in ascending order every time an input blurs
+  const handleThresholdBlur = () => {
+    if (steps !== undefined) {
+      const sortedSteps = [...steps];
+      sortedSteps.sort((a, b) => a.value - b.value);
+      if (thresholds !== undefined) {
+        onChange(
+          produce(thresholds, (draft) => {
+            draft.steps = sortedSteps;
+          })
+        );
+      }
+    }
+  };
+
+  const deleteThreshold = (i: number): void => {
+    if (thresholds !== undefined) {
+      const updatedThresholds = produce(thresholds, (draft) => {
+        if (draft.steps) {
+          draft.steps.splice(i, 1);
+        }
+      });
+      onChange(updatedThresholds);
+    }
+  };
+
+  const addThresholdInput = (): void => {
+    focusRef.current = true;
+    if (thresholds === undefined) {
+      onChange({
+        steps: [{ value: 10 }],
+      });
+    } else {
+      onChange(
+        produce(thresholds, (draft) => {
+          const steps = draft.steps;
+          if (steps?.length) {
+            const lastStep = steps[steps.length - 1];
+            const color = palette[steps.length] ?? getRandomColor(); // we will assign color from the palette first, then generate random color
+            steps.push({ color, value: (lastStep?.value ?? 0) + 10 });
+          } else if (steps) {
+            steps.push({ value: 10 });
```

Comment:
Is this saying that any newly added threshold defaults to a value of `10`?

Reply:
if there is no previous threshold set, then the first threshold you add will default to `10`